### DISCUSSION TITLE
How to conditionally update sub-shapes in G6 v5 when overriding the drawOtherShapes method

### DISCUSSION THREAD

**Discussion by Chrishaneus** [antvis/G6#5537] [Q&A]
![image](https://github.com/antvis/G6/assets/91400728/4fd2bfb4-5bcb-476d-ab99-29f6133316f1)

I am currently working on a project that visualizes systems such as the figure shown above. I have seen that rendering images takes a lot of toll memory-wise so I was wondering if there are ways to conditionally draw shapes every time a node updates (particularly through the graph.updateData method). I am currently using g6 v5.0.0 beta.28.

In version 4, there are methods when registering nodes that we can specifically specify so that we can conditionally update the group shapes.

```
G6.registerNode('new-node-type', {
    {...config},
    update: (cfg, item) => {
      ...update method
    },
    drawShape: (cfg, group) => {
      return ...
    },
    setState: (name, value, item) => {
      return ...
    },
  });
```

**Comment by Aarebecca** [comment]
I'm sorry that it is not currently possible to control the drawing conditions of sub-shapes. However, we are currently restructuring the logic of element drawing. The new plan has already taken this issue into consideration, and we will provide a solution to the above problem in the future.
